lib/Db/PimcoreExtensionsTrait.php

```diff
@@ -340,22 +340,25 @@ public function fetchPairs($sql, array $params = [], $types = [])
      */
     public function insertOrUpdate($table, array $data)
     {
+        // get the field name of the primary key
+        $tableDetails = $this->getSchemaManager()->listTableDetails($table);
```

Comment:
I'm afraid we need a cache here, just did a quick benchmark and it seems this values are not getting cached by Doctrine. I'll merge it now anyway and create a follow-up PR with the cache. Thanks a lot!